core/capabilities/remote/executable/request/client_request.go

```diff
@@ -88,9 +93,43 @@ func newClientRequest(ctx context.Context, lggr logger.Logger, requestID string,
 	}
 
 	responseReceived := make(map[p2ptypes.PeerID]bool)
-	ctxWithCancel, cancelFn := context.WithCancel(ctx)
 
-	lggr.Debugw("sending request to peers", "schedule", peerIDToTransmissionDelay)
+	maxDelayDuration := time.Duration(0)
+	for _, delay := range peerIDToTransmissionDelay {
+		if delay > maxDelayDuration {
+			maxDelayDuration = delay
+		}
+	}
+
+	// Add some margin to allow the last peer to respond
+	maxDelayDuration += defaultDelayMargin
+
+	// Instantiate a new context based on the parent, but without its deadline.
+	// We set a new deadline instead equal to the original timeout OR the full length
+	// of the execution schedule plus some margin, whichever is greater
+
+	// We do this to ensure that we will always execute the entire transmission schedule.
+	// This ensures that all capability DON nodes will receive a quorum of requests,
+	// and will execute all requests they receive from the workflow DON, preventing
+	// quorum errors from lagging members of the workflow DON.
+	dl, ok := ctx.Deadline()
+	originalTimeout := time.Duration(0)
+	if ok {
+		originalTimeout = time.Until(dl)
+	}
+	effectiveTimeout := originalTimeout
+	if originalTimeout < maxDelayDuration {
+		effectiveTimeout = maxDelayDuration
+	}
+
+	// Now let's create a new context based on the adjusted timeout value.
+	// By calling WithoutCancel, we ensure that this context can only be cancelled in
+	// one of two ways -- 1) by explicitly calling the cancelFn we create below, or 2)
+	// after the adjusted timeout expires.
+	ctxWithoutCancel := context.WithoutCancel(ctx)
+	ctxWithCancel, cancelFn := context.WithTimeout(ctxWithoutCancel, effectiveTimeout)
```

Comment:
the closure below waits on `ctxWithCancel.Done()` but accepts `ctx` as an argument, this mixes closure versus passed context.

Also, where is `ctxWithoutCancel` used?